### PR TITLE
Don't encode `product_key` as part of the dialog api url path.

### PR DESCRIPTION
**Pull request by suligap**
Instead add it as any other param (as a url query parameter).

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -98,5 +103,44 @@ def test_failure_url_param(self):
         url = self.lp.get_buy_url(item, failure_url="http://example.com")
         self.assertTrue('failure_url' in url)
 
+    def test_get_web_url_product_key_param(self):
+        item = ItemDefinition(1, 'EUR20', 'DE19.0', 'http://help.me/', 'title')
```
Comment (doismellburning) [inline tests/test_client.py]: While I'm sure http://help.me/ is going to be a fascinating site, would you mind just using example.com please? ;)

File: tests/test_client.py [hunk 2]
```diff
@@ -42,6 +44,9 @@ def get_qs_dict(self, url):
         d = parse_qs(o.query)
         return d
 
+    def get_dialog_api_furl(self, url):
+        return furl(furl(url).query.params['url'])
```
Comment (doismellburning) [inline tests/test_client.py]: ...there's totally a meme here ;)

**Comment by doismellburning** [comment]
Sorry - should have commented this earlier - would you mind sticking explanatory docstrings in the tests please - I think without that they're potentially a bit opaque to anyone who doesn't have this PR's context!

Otherwise, looks great, cheers :)

**Comment by suligap** [comment]
Thanks! Good point. Added docstrings.

**Pull request merged**